### DISCUSSION TITLE
How to use the diff exit codes in the cli?

### DISCUSSION THREAD

**Discussion by pkwiatkowski0** [aconrad/pycobertura#157] [Q&A] [answered]
Hi @aconrad, 
I was searching a tool for coverage comparison for my organization. PyCobertura seems to provide what we need, especially the exit code 2:
> 2: the changes worsened the overall coverage

For the following example case:
![image](https://user-images.githubusercontent.com/24497498/192985036-9c228903-d0c0-4e64-916c-d27a7602cf9e.png)
where ` one-more-line-coverage.xml` differs from the previous with one uncovered line added. 
I just wonder how to catch this code in the terminal. Ultimately I want to use exit code 2  to fail the gitlab's pipeline.

P.S. Looking forward for the `--format json` feature.
Best regards

**Comment by aconrad** [comment, marked as answer]
The exit code is what the command returns on the shell. You can access the exit code by typing `$?` right after running pycobertura, it's the syntax for getting the exit code of the last command run.

```bash
pycobertura ...
PYCOBERTURA_EXIT_CODE=$?
echo $PYCOBERTURA_EXIT_CODE
```

Let me know if that answers your questions.

**Reply by pkwiatkowski0** [reply]
ok, thank you very much, now I can use it for a better error explanation.

**Reply by aconrad** [reply]
I also updated the README with the example above.

https://github.com/aconrad/pycobertura/blob/master/README.md#diff-exit-codes
